src/main/features/claim/routes/pay.ts

```diff
@@ -85,27 +85,29 @@ export default express.Router()
         throw new Error('No amount entered, you cannot pay yet')
       }
 
-      const paymentId = draft.document.claimant.payment.id
+      const paymentRef = draft.document.claimant.payment ? draft.document.claimant.payment.reference : undefined
 
-      if (paymentId) {
+      if (paymentRef) {
         const payClient: PayClient = await getPayClient()
-        const payment: Payment = await payClient.retrieve(user, paymentId)
-        switch (payment.state.status) {
-          case 'success':
-            return res.redirect(Paths.finishPaymentReceiver.evaluateUri({ externalId: draft.document.externalId }))
+        const paymentResponse: PaymentRetrieveResponse = await payClient.retrieve(user, paymentRef)
+        if (paymentResponse !== undefined) {
+          switch (paymentResponse.status) {
+            case 'Success':
+              return res.redirect(Paths.finishPaymentReceiver.evaluateUri({ externalId: draft.document.externalId }))
+          }
+        } else if (draft.document.claimant.payment['state'] && draft.document.claimant.payment['state']['status'] === 'success') {
```

Comment:
That gives us some logs but if draft failed before updated version was saved in the draft store we will not have that log :(